plugin/src/main/java/org/opensearch/ml/action/update/UpdateModelTransportAction.java

```diff
@@ -0,0 +1,564 @@
+                    client.update(updateRequest, getUpdateResponseListener(modelId, wrappedListener));
+                }, e -> {
+                    log
+                        .error(
+                            "Failed to register ML model with model ID {} to the new model group with model group ID {}",
+                            modelId,
+                            newModelGroupId,
+                            e
+                        );
+                    wrappedListener.onFailure(e);
+                }));
+            }
+        } catch (IOException e) {
+            log.error("Failed to build update request.");
+            wrappedListener.onFailure(e);
+        }
+    }
+
+    private ActionListener<UpdateResponse> getUpdateResponseListenerWithUpdateModelCache(
+        String modelId,
+        ActionListener<UpdateResponse> wrappedListener,
+        MLUpdateModelCacheNodesRequest mlUpdateModelCacheNodesRequest
+    ) {
+        return ActionListener.wrap(updateResponse -> {
+            // The update response returned an unexpected status may indicate a failed update
+            if (updateResponse != null && updateResponse.getResult() != DocWriteResponse.Result.UPDATED) {
+                log.error("Model id:{} failed update", modelId);
+                wrappedListener.onResponse(updateResponse);
+                return;
+            }
+            client.execute(MLUpdateModelCacheAction.INSTANCE, mlUpdateModelCacheNodesRequest, ActionListener.wrap(r -> {
+                log.info("Successfully perform in-place update ML model with model ID {}", modelId);
+                wrappedListener.onResponse(updateResponse);
+            }, e -> {
+                log.error("Failed to perform in-place update for model: {}" + modelId, e);
+                wrappedListener.onFailure(e);
+            }));
+        }, exception -> {
+            log.error("Failed to update ML model: " + modelId, exception);
+            wrappedListener.onFailure(exception);
+        });
+    }
+
+    private ActionListener<UpdateResponse> getUpdateResponseListener(String modelId, ActionListener<UpdateResponse> wrappedListener) {
+        return ActionListener.wrap(updateResponse -> {
+            if (updateResponse != null && updateResponse.getResult() != DocWriteResponse.Result.UPDATED) {
+                log.info("Model id:{} failed update", modelId);
+                wrappedListener.onResponse(updateResponse);
+                return;
+            }
+            log.info("Successfully update ML model with model ID {}", modelId);
+            wrappedListener.onResponse(updateResponse);
+        }, exception -> {
+            log.error("Failed to update ML model: " + modelId, exception);
+            wrappedListener.onFailure(exception);
+        });
+    }
+
+    private String incrementLatestVersion(Map<String, Object> modelGroupSourceMap) {
+        return Integer.toString((int) modelGroupSourceMap.get(MLModelGroup.LATEST_VERSION_FIELD) + 1);
```

Comment:
Seems can't handle race condition. What if two users get same latest version field at same time and try to increment to create new version? Refer to register model API

Reply:
Following the model group registration pattern in model registration can resolve this issue.